tools/code-analyzer/src/git.ts

```diff
@@ -53,13 +53,38 @@ export const cloneRepo = async ( repoPath: string ) => {
 	return folderPath;
 };
 
+export const sparseCheckoutRepo = async (
```

Comment:
One thing I wonder is, why do we have to JS doc our TypeScript? I'd rather avoid the extra work. I'll add these for now but its annoying.